crates/shared/src/price_estimation/trade_verifier.rs

```diff
@@ -358,10 +393,38 @@ fn encode_settlement(
         tracing::trace!("adding unwrap interaction for paying out ETH");
     }
 
-    let tokens = vec![query.sell_token, query.buy_token];
-    let clearing_prices = match query.kind {
-        OrderKind::Sell => vec![trade.out_amount, query.in_amount.get()],
-        OrderKind::Buy => vec![query.in_amount.get(), trade.out_amount],
+    let mut tokens = vec![query.sell_token, query.buy_token];
+    let mut clearing_prices = match (trade, query.kind) {
+        (Trade::Legacy(_), OrderKind::Sell) => {
+            vec![*out_amount, query.in_amount.get()]
+        }
+        (Trade::Legacy(_), OrderKind::Buy) => {
+            vec![query.in_amount.get(), *out_amount]
+        }
+        (Trade::WithJitOrders(trade), OrderKind::Sell) => {
+            vec![
+                *trade
+                    .clearing_prices
+                    .get(&query.sell_token)
+                    .context("sell token clearing price is missing")?,
+                *trade
+                    .clearing_prices
+                    .get(&query.buy_token)
+                    .context("buy token clearing price is missing")?,
+            ]
+        }
+        (Trade::WithJitOrders(trade), OrderKind::Buy) => {
```

Comment:
When the solver actually reports the clearing price vector we don't have to switch around the order prices for buy orders. The solver needs to return the actual prices for the given tokens.